grpc.go

```diff
@@ -718,44 +718,100 @@ func (a *agentGRPC) DestroySandbox(ctx context.Context, req *pb.DestroySandboxRe
 	return emptyResp, nil
 }
 
-func (a *agentGRPC) AddInterface(ctx context.Context, req *pb.AddInterfaceRequest) (*gpb.Empty, error) {
-	if err := a.sandbox.addInterface(nil, req.Interface); err != nil {
-		return emptyResp, err
-	}
+func (a *agentGRPC) AddInterfaces(ctx context.Context, req *pb.AddInterfacesRequest) (*pb.Interfaces, error) {
 
-	return emptyResp, nil
+	var resultingInterfaces pb.Interfaces
+
+	for _, ifc := range req.Interfaces {
+		if ifc == nil {
+			continue
+		}
+
+		err := a.sandbox.addInterface(nil, ifc)
+		if err != nil {
+			return &resultingInterfaces, err
+		}
+
+		resultingInterfaces.Interfaces = append(resultingInterfaces.Interfaces, ifc)
+	}
+	return &resultingInterfaces, nil
 }
 
-func (a *agentGRPC) RemoveInterface(ctx context.Context, req *pb.RemoveInterfaceRequest) (*gpb.Empty, error) {
-	if err := a.sandbox.removeInterface(nil, req.Name); err != nil {
-		return emptyResp, err
+func (a *agentGRPC) UpdateInterfaces(ctx context.Context, req *pb.UpdateInterfacesRequest) (*pb.Interfaces, error) {
+
+	var resultingInterfaces pb.Interfaces
+
+	for _, ifc := range req.Interfaces {
+		if ifc == nil {
+			continue
+		}
+
+		err := a.sandbox.updateInterface(nil, ifc)
+		if err != nil {
+			return &resultingInterfaces, err
+		}
+
+		resultingInterfaces.Interfaces = append(resultingInterfaces.Interfaces, ifc)
 	}
 
-	return emptyResp, nil
+	return &resultingInterfaces, nil
 }
 
-func (a *agentGRPC) UpdateInterface(ctx context.Context, req *pb.UpdateInterfaceRequest) (*gpb.Empty, error) {
-	if err := a.sandbox.updateInterface(nil, req.Interface, req.Type); err != nil {
-		return emptyResp, err
+func (a *agentGRPC) RemoveInterfaces(ctx context.Context, req *pb.RemoveInterfacesRequest) (*pb.Interfaces, error) {
+
+	var resultingInterfaces pb.Interfaces
+
+	resultingInterfaces.Interfaces = req.Interfaces
+
+	for _, ifc := range req.Interfaces {
+		if ifc == nil {
+			continue
+		}
+
+		err := a.sandbox.removeInterface(nil, ifc)
+		if err != nil {
+			return &resultingInterfaces, err
+		}
+		resultingInterfaces.Interfaces = append(resultingInterfaces.Interfaces[:0], resultingInterfaces.Interfaces[1:]...)
```

Comment:
This logic is wrong when the req.Interfaces = {A, nil, B} because `if ifc == nil {continue}`

This is removing always the first interface from list and it's not correct

And it's actually equal to 
`resultingInterfaces.Interfaces = resultingInterfaces.Interfaces[1:]`